lakefile.lean

```diff
@@ -2,13 +2,26 @@ import Lake
 
 open Lake DSL
 
+/-- Turn a config option into a Boolean -/
+def enabled (val : Option String) (default : Bool := false) : Bool :=
+  match val with
+  | .some "off" => false
+  | .some "false" => false
+  | .some "on" => true
+  | .some "true" => true
+  | .some _ => not default -- This allows -K=opt to flip the default.
+  | _ => default
+
+-- Selectively choose an option if feature is enabled.
+macro "optArg?" x:ident ws d:ident v:term : term =>
+  `(if enabled (get_config? $x) $d then $v else default)
+
 package std where
-  moreLeanArgs := #["-DwarningAsError=true"]
-  leanOptions := Id.run do
-    let mut opts := #[⟨`linter.missingDocs, true⟩]
-    if get_config? disable_new_compiler |>.isSome then
-      opts := opts.push ⟨`compiler.enableNew, false⟩
-    opts
+  moreLeanArgs :=
+    optArg? werror false #["-DwarningAsError=true"]
```

Comment:
I would also like lake to gain specific knowledge of this flag so that it can be controlled without having to reimplement this logic in every project that uses CI